geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java

```diff
@@ -533,7 +535,8 @@ public boolean checkMyStateOnMembers(Set<InternalDistributedMember> replicates)
         String message = String.format(
             "Region %s remote member %s with persistent data %s was not part of the same distributed system as the local data from %s",
             regionPath, member, remoteId, myId);
-        throw new ConflictingPersistentDataException(message);
+        logger.warn(message);
+        iterator.remove();
```

Comment:
Is synchronization neededhere? The `stateOnPeers` maybe shared with other threads.

Reply:
it won't. This data structure is in reply message. Only the message processor will use it